cancer_ai/validator/model_db.py

```diff
@@ -0,0 +1,184 @@
+                    hotkey = hotkey
+                )
+                session.add(model_record)
+                session.commit()
+                bt.logging.debug(f"Successfully added model info for hotkey {hotkey} into the DB.")
+            except Exception as e:
+                session.rollback()
+                raise e
+            finally:
+                session.close()
+        else:
+            bt.logging.debug(f"Model for hotkey {hotkey} and date {date_submitted} already exists, skipping.")
+    
+    def get_model(self, date_submitted: datetime, hotkey: str):
+        session = self.Session()
+        try:
+            model_record = session.query(ChainMinerModelDB).filter_by(
+                date_submitted=date_submitted, hotkey=hotkey
+            ).first()
+            if model_record:
+                return ChainMinerModel(
+                    competition_id = model_record.competition_id,
+                    hf_repo_id = model_record.hf_repo_id,
+                    hf_model_filename = model_record.hf_model_filename,
+                    hf_repo_type = model_record.hf_repo_type,
+                    hf_code_filename = model_record.hf_code_filename,
+                    block = model_record.block,
+                    hotkey = model_record.hotkey
+                )
+            return None
+        finally:
+            session.close()
+
+    def delete_model(self, date_submitted: datetime, hotkey: str):
+        session = self.Session()
+        try:
+            model_record = session.query(ChainMinerModelDB).filter_by(
+                date_submitted=date_submitted, hotkey=hotkey
+            ).first()
+            if model_record:
+                session.delete(model_record)
+                session.commit()
+                return True
+            return False
+        except Exception as e:
+            session.rollback()
+            raise e
+        finally:
+            session.close()
+
+    def get_block_timestamp(self, block_number):
+        """Gets the timestamp of a block given its number."""
+        try:
+            block_hash = self.subtensor.get_block_hash(block_number)
+            if block_hash is None:
+                raise ValueError(f"Block hash not found for block number {block_number}")
+            
+            timestamp_info = self.subtensor.substrate.query(
+                module='Timestamp',
+                storage_function='Now',
+                block_hash=block_hash
+            )
+
+            if timestamp_info is None:
+                raise ValueError(f"Timestamp not found for block hash {block_hash}")
+
+            timestamp_ms = timestamp_info.value
+            block_datetime = datetime.fromtimestamp(timestamp_ms / 1000.0)
+
+            return block_datetime
+        except Exception as e:
+            bt.logging.error(f"Error retrieving block timestamp: {e}")
+            raise
+
+    def get_latest_models(self, hotkeys: list[str]) -> list[ChainMinerModel]:
```

Comment:
merge with `get_model`, at least use `get_model` inside this method

Reply:
get_model requires date_submitted whereas get_latest_models is not supposed to be called with those parameters, its figuring it out programatically